src/arraymancer/tensor/private/p_display.nim

```diff
@@ -215,4 +287,136 @@ proc prettyImpl*[T](t: Tensor[T],
   else:
     result = t.disp2d(alignBy = alignBy,
                       alignSpacing = alignSpacing,
-                      precision = precision).strip
+                      precision = precision,
+                      specifier = specifier).strip
+
+proc disp1dAsArray[T](t: Tensor[T],
+                    sep = ", ",
+                    precision = -1, specifier: static string = ""): string =
+  ## Display a 1D-tensor (only used for "array-style", i.e. non-table, printing)
+  if t.len == 0:
+    return "[]"
+  result = "["
+  for value in t:
+    result &= dispElement(value, precision = precision, specifier = specifier)
+    result &= sep
+  # Remove the separator from the last element
+  result = result[0..^(1+sep.len)] & "]"
+  when T is Complex and "j" in specifier:
+    result = result.replace("(").replace(")")
+
+proc compactTensorDescription[T](t: Tensor[T]): string =
+  ## Describe the tensor in terms of its shape and type (in a "compact" way)
+  ## Only used for array-style printing
+  # Most if not all tensors element types are part of the system or complex
+  # modules so we can remove them from the type without must information loss
```

Comment:
```suggestion
  # modules so we can remove them from the type without much information loss
```

Reply:
Will fix this typo in a commit taking care of all your comments.